src/EditorFeatures/Core.Cocoa/Workspaces/CocoaTextBufferVisibilityTracker.cs

```diff
@@ -0,0 +1,225 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Workspaces
+{
+    [Export(typeof(ITextBufferVisibilityTracker))]
+    internal sealed class CocoaTextBufferVisibilityTracker : ITextBufferVisibilityTracker
```

Comment:
i haven't looked closely.  but the presumption is that this is a straight port, just tweaked fort he cocoa visibility checks.  if so, this is ok with me.  if anything else was changed, lmk.  thanks!

Reply:
Yup, that is the intent! Thank you!

Reply:
Yeah, it's just the commit I initially made I think. And that was the changeset, I think the diff itself is just removing `.VisualElement` and changing the event handler type.